core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/state/RepairHistoryProviderImpl.java

```diff
@@ -49,28 +51,34 @@
 
     private static final String REPAIR_HISTORY_BY_TIME_STATEMENT = String
             .format("SELECT id, range_begin, range_end, status, participants FROM %s.%s WHERE keyspace_name=? AND columnfamily_name=? AND id >= minTimeuuid(?) and id <= maxTimeuuid(?)", KEYSPACE_NAME, REPAIR_HISTORY);
-    private static final String REPAIR_HISTORY_STATEMENT = String.format("SELECT id, range_begin, range_end, status, participants FROM %s.%s WHERE keyspace_name=? AND columnfamily_name=? AND id <= maxTimeuuid(?)", KEYSPACE_NAME, REPAIR_HISTORY);
 
     private final Session mySession;
     private final StatementDecorator myStatementDecorator;
 
     private final PreparedStatement myRepairHistoryByTimeStatement;
-    private final PreparedStatement myRepairHistoryStatement;
+    private final long myLookbackTime;
+    private final Clock myClock;
 
-    public RepairHistoryProviderImpl(Session session, StatementDecorator statementDecorator)
+    public RepairHistoryProviderImpl(Session session, StatementDecorator statementDecorator, long lookbackTime)
+    {
+        this(session, statementDecorator, lookbackTime, Clock.systemDefaultZone());
+    }
+
+    @VisibleForTesting
+    RepairHistoryProviderImpl(Session session, StatementDecorator statementDecorator, long lookbackTime, Clock clock)
     {
         mySession = session;
         myStatementDecorator = statementDecorator;
         myRepairHistoryByTimeStatement = mySession.prepare(REPAIR_HISTORY_BY_TIME_STATEMENT);
-        myRepairHistoryStatement = mySession.prepare(REPAIR_HISTORY_STATEMENT);
+        myLookbackTime = lookbackTime;
+        myClock = clock;
     }
 
     @Override
     public Iterator<RepairEntry> iterate(TableReference tableReference, long to, Predicate<RepairEntry> predicate)
     {
-        ResultSet resultSet = execute(myRepairHistoryStatement.bind(tableReference.getKeyspace(), tableReference.getTable(), new Date(to)));
-
-        return RepairEntryIterator.create(resultSet.iterator(), predicate);
+        long from = myClock.millis() - myLookbackTime;
```

Comment:
This could potentially be a (theoretical) problem... IF myLookbackTime is a huge number (e.g. Long.MAX_VALUE), then java.util.Date(from) will be in the future (instead of being far in the past)... Maybe that is not a problem in practice?
But if we should handle this - what is the from-value we should limit to?
0? - in that case we cannot find entries being created before epoch (1970), but that is maybe OK...

Reply:
I'd be ok with introducing a check for it (in the next method), something like:

```
if (!startDate.before(endDate))
{
  throw new IllegalArgumentException("");
}
```

But then we should add a test case for it as well.

---

Also I think this could be an issue if "to" is sent in lower than "from".

Reply:
added check